### ISSUE TITLE
[chains] potential CHAIN_ID problem results in invalid signature error from 'eris pkgs'

### ISSUE THREAD

**Issue by j-h-scheufen** [monax/cli#1133]
This ticket is the conclusion of and replaces the following issues that were initially thought to be rooted in other repos.

https://github.com/eris-ltd/eris-db/issues/352
https://github.com/eris-ltd/eris-keys/issues/90

### Requirements
- ability to instantiate a chain from an existing configuration (via `--init-dir`) under a chain ID that differs from the one used in the `make` operation.

### Problem
In the bundles repo the test.sh script which was failing since the 0.12.0 release uses a saved chain configuration created with the chain ID "bundles-test".  However, when starting a chain from that config, a generated chain ID is used (`bundles-test-$uuid`).
The chain seems to start fine and produces blocks. Calling `inspect` reveals that the `CHAIN_ID` and `CHAIN_NAME` env parameters are set as expected:
```
Env:                
                    CHAIN_ID=test1
                    CHAIN_NAME=test1
                    ERIS_DB_WORKDIR=/home/eris/.eris/chains/test1
                    CONTAINER_NAME=test1-94a563ec-c12e-47ec-a1ae-3dd571c9edaa
                    CHAINID=test1
                    PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
                    GO15VENDOREXPERIMENT=1
                    GOPATH=/go
                    INSTALL_BASE=/usr/local/bin
                    USER=eris
                    ERIS=/home/eris/.eris
                    TARGET=eris-db
```
However, when checking in the log, the chain's ID still shows up as the originally created one:
```
NOTE[12-09|21:41:55] Finalizing commit of block with 0 txs    module=consensus height=13 hash=F21F22A5C5E428341139EDC17213D5C3FE356E87
INFO[12-09|21:41:55] Block{
  Header{
    ChainID:        bundles-test
    Height:         13
    Time:           2016-12-09 21:41:55.657 +0000 UTC
    NumTxs:         0
    LastBlock:      7B179ABD864A75D4A2862993D3EE7FD8EB672E7E
    LastBlockParts: PartSet{T:1 6E78BDF1E77B}
    LastCommit:     1FEAB2CCC60E0FB262FCD7005E2A581A1FEE7CC3
    Data:           
    Validators:     C29D8F0AD231BFAE54004DE78ACCB637023C376D
    App:            F2841730F774F339BEED35B484EA9E6FED7D6BFF
  }#F21F22A5C5E428341139EDC17213D5C3FE356E87
  Data{
    
  }#
  Commit{
    Precommits: Vote{12/00/2(Precommit) 7B179ABD864A#PartSet{T:1 6E78BDF1E77B} /1F1686054EB5.../}
  }#1FEAB2CCC60E0FB262FCD7005E2A581A1FEE7CC3
}#F21F22A5C5E428341139EDC17213D5C3FE356E87 module=consensus
```

This situation is believed to somehow be the root cause of a "invalid signature" error in our test script when interacting with a chain with a generated ID:
```
Executing Job                                 defaultAddr
Executing Job                                 BaseErrorsLib
Deploying Contract                       name=BaseErrors
Encountered error waiting for event: timed out waiting for event

                                      chainID=bundles-test-efe5922e9153
                                     chainURL=tcp://chain:46657
                                      defAddr=3590E8D7753E5D95A3CA9DA094E804249D3365BC
                                       rawErr=Internal error: Error invalid signature
Error finalizing contract deploy contract_bundles/erisindustries/commons-base/BaseErrors.sol: 
There has been an error talking to your eris chain.

Internal error: Error invalid signature

Debugging this error is tricky, but don't worry the marmot recovery checklist is...
  * is the 3590E8D7753E5D95A3CA9DA094E804249D3365BC account right?
  * is the account you want to use in your keys service: eris keys ls ?
  * is the account you want to use in your genesis.json: eris chains cat bundles-test-efe5922e9153 genesis ?
  * is your chain making blocks: eris chains logs -f bundles-test-efe5922e9153 ?
  * do you have permissions to do what you're trying to do on the chain?
```

To reproduce:
```
$> eris chains make test --account-types Full:1,Participant:10
$> eris chains start newName --init-dir ~/.eris/chains/test/test_full_000
$> eris pkgs do -c newName -a (key from ~/.eris/chains/test/test_full_000 account)
```

**Comment by zramsay** [comment]
this will be resolved by #858

**Comment by zramsay** [comment]
closing since #858 will address this
